homeassistant/components/medcom_ble/__init__.py

```diff
@@ -0,0 +1,73 @@
+"""The Medcom BLE integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from medcom_ble import MedcomBleDeviceData
+
+from homeassistant.components import bluetooth
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util.unit_system import METRIC_SYSTEM
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN
+
+# Supported platforms
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Medcom BLE radiation monitor from a config entry."""
+
+    address = entry.unique_id
+    elevation = hass.config.elevation
+    is_metric = hass.config.units is METRIC_SYSTEM
+    assert address is not None
+
+    ble_device = bluetooth.async_ble_device_from_address(hass, address)
+    if not ble_device:
+        raise ConfigEntryNotReady(
+            f"Could not find Medcom BLE device with address {address}"
+        )
+
+    async def _async_update_method():
+        """Get data from Medcom BLE radiation monitor."""
+        ble_device = bluetooth.async_ble_device_from_address(hass, address)
+        inspector = MedcomBleDeviceData(_LOGGER, elevation, is_metric)
+
+        try:
+            data = await inspector.update_device(ble_device)
+        except Exception as err:
```

Comment:
We don't want a broad exception catch outside of the config flow as it hides bugs. Please only catch exceptions that are expected here

Reply:
Sounds good. Shall I also aim to open a PR against the `airthings_ble` integration as well to refactor it the same way? Most of the comments on this PR apply to the existing code on many integration...

Reply:
Looking closer, I can definitely narrow down the Exceptions that will be caught -  `update_device` should only raise a `BleakError` , but this `except Exception as err:` during device updates/connections to devices is a pattern that is widely used across integrations on HA - about 50 different integrations after a first look. Is this something you expect to fix across all integrations, or is there something special about this particular integration I'm missing?

Reply:
> but this `except Exception as err:` during device updates/connections to devices is a pattern that is widely used across integrations on HA - about 50 different integrations after a first look. Is this something you expect to fix across all integrations, or is there something special about this particular integration I'm missing?

This should be fixed across all integrations eventually, we only allow broad exception catches in config flow for new code.

Reply:
> but this `except Exception as err:` during device updates/connections to devices is a pattern that is widely used across integrations on HA - about 50 different integrations after a first look. Is this something you expect to fix across all integrations, or is there something special about this particular integration I'm missing?

This should be fixed across all integrations eventually, we only allow broad exception catches in config flow for new code.